### PR TITLE
make sure the bit width is propper

### PR DESCRIPTION
**Pull request by dothebart**
### Scope & Purpose

https://github.com/arangodb/arangodb/issues/17756 indicates that we may have portability issues here

- [x] :hankey: Bugfix

### CODE REVIEW

File: 3rdParty/jemalloc/v5.3.0/src/pages.c [hunk 1]
```diff
@@ -579,7 +579,7 @@ os_page_detect(void) {
 	return getpagesize();
 #else
 	long result = sysconf(_SC_PAGESIZE);
-	if (result == -1) {
```
Comment (MBkkt) [inline 3rdParty/jemalloc/v5.3.0/src/pages.c:582]: it will cast implicit to long

File: 3rdParty/jemalloc/v5.3.0/configure.ac [hunk 2]
```diff
@@ -1719,7 +1719,7 @@ if test "x$LG_PAGE" = "xdetect"; then
 #else
     result = sysconf(_SC_PAGESIZE);
 #endif
-    if (result == -1) {
+    if (result == (long) -1) {
```
Comment (MBkkt) [inline 3rdParty/jemalloc/v5.3.0/configure.ac:1722]: it will cast implicit to long

File: 3rdParty/jemalloc/v5.3.0/configure [hunk 3]
```diff
@@ -11681,7 +11681,7 @@ main ()
 #else
     result = sysconf(_SC_PAGESIZE);
 #endif
-    if (result == -1) {
+    if (result == (long) -1) {
```
Comment (MBkkt) [inline 3rdParty/jemalloc/v5.3.0/configure:11684]: it will cast implicit to long

**Comment by dothebart** [comment]
closing as won't fix.

**Pull request closed without merging**